### PR TITLE
add a description field in TooManyIncorrectAuthorities

### PR DESCRIPTION
**Pull request by longbowlu**
we see `TooManyIncorrectAuthorities` error here and there but more times than not we don't know for what actions the authority aggregator said so. This PR adds an additional field `action` to describe it

### CODE REVIEW

File: crates/sui-core/src/authority_aggregator.rs [hunk 1]
```diff
@@ -871,6 +874,7 @@ where
                                             response.err().map(|err| (name, err))
                                         })
                                         .collect(),
+                                    action: "get_object_by_id".to_string(),
```
Comment (velvia) [inline crates/sui-core/src/authority_aggregator.rs:877]: Is there a way the action field can be a `&str`?  You see this is very inefficient, in that we have to create a new string copy for what should be a CONST or static string.

I mean I know this is just error, but trying to be mindful of where we can save allocations.  :)

**Review by velvia — CHANGES_REQUESTED** [review]
Thanks for doing this.  If you can change the error that would be great.  Or use something like `FlexStr`, which we use for event reporting, as it can avoid allocations for static error strings.

**Review by mwtian — APPROVED** [review]

File: crates/sui-core/src/authority_aggregator.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-core/src/authority_aggregator.rs]: replace with `&'static str`

**Review by velvia — APPROVED** [review]
Thanks

**Pull request merged**